redbot/core/bot.py

```diff
@@ -1145,9 +1145,26 @@ async def _pre_login(self) -> None:
             self.owner_ids.add(self._owner_id_overwrite)
 
         i18n_locale = await self._config.locale()
-        _i18n.set_global_locale(i18n_locale)
+        try:
+            _i18n.set_global_locale(i18n_locale)
+        except ValueError:
+            i18n_locale = _i18n.FRESH_INSTALL_LOCALE
+            await self._config.locale.clear()
+            log.warning(
+                "The bot's global locale was set to an invalid value and has been reset"
```

Comment:
I probably would like to see it print the old and incorrect value in the warning message for debugging purposes - We don't want a case in the future where this catch bugs out to confuse users if their language gets reset.

Reply:
Updated.